### PR TITLE
:memo: matcher documentation

### LINKED ISSUE #297: give `store` to `matcher` (if not already the case)
So a user can do
`when((action, store) => action.type === store.data.shows.SET)`

### PR DESCRIPTION
**Pull request by fabienjuif**
fixes #297 

FYI @EmrysMyrddin ;)

### CODE REVIEW

File: packages/k-ramel/API.md [hunk 1]
```diff
@@ -17,3 +17,83 @@ All options are optionals to keep the default usage as simple as possible.
...
+  store.show.set(action.payload)
+})
+```
+
+#### Example 4: with a function, use the store
+In the function matcher you have access to the store.
+It can be useful when you want to match a specific Redux action since our actions types are published. (if you use `types`).
+
+```js
+when((action, store) => action.type === store.show.SET)(() => { console.log('show is set!') })
```
Comment (EmrysMyrddin) [inline packages/k-ramel/API.md]: It's cool to see that it's already possible to do it :-)

I will try to use k-ramel for a while in a project and I will make a proposition if I find it useful to add a custom matcher for a better readability.

Comment (fabienjuif) [reply packages/k-ramel/API.md]: thank you

Comment (fabienjuif) [reply packages/k-ramel/API.md]: If we add an helper, we can add it in a minor version

File: packages/k-ramel/API.md [hunk 2]
```diff
@@ -17,3 +17,83 @@ All options are optionals to keep the default usage as simple as possible.
...
+})
+```
+
+#### Exemple 2: with a regexp
+In this example we react to the http driver response to retrieve **Adventure time** show description that the API responds.
+We take care of looking for action.status so we make sure the request is ok!
+
+```js
+when(/@@http\/.*TVMAZE.*/)((action, store) => {
+  // if the request is failed, we are doing nothing
```
Comment (delphinemillet) [inline packages/k-ramel/API.md]:
```suggestion
  // if the request fails, we do nothing
```

File: packages/k-ramel/API.md [hunk 3]
```diff
@@ -17,3 +17,83 @@ All options are optionals to keep the default usage as simple as possible.
...
+  if (action.status < 200 || action.status >= 400) return
+
+  // in others cases we register our show description to the store
```
Comment (delphinemillet) [inline packages/k-ramel/API.md]:
```suggestion
  // in other cases we register our show description to the store
```

File: packages/k-ramel/API.md [hunk 4]
```diff
@@ -17,3 +17,83 @@ All options are optionals to keep the default usage as simple as possible.
...
+  store.show.set(action.payload)
+})
+```
+
+#### Example 3: with a function
+We are doing the same example given previously but using a function matcher.
+ - The first matcher used is a regexp (like previous exemple): we are making sure to process a HTTP action.
```
Comment (delphinemillet) [inline packages/k-ramel/API.md]:
```suggestion
 - The first matcher used is a regexp (like previous exemple): we ensure to process a HTTP action.
```

File: packages/k-ramel/API.md [hunk 5]
```diff
@@ -17,3 +17,83 @@ All options are optionals to keep the default usage as simple as possible.
...
+ - The second matcher is a function to make sure the request is not failed!
```
Comment (delphinemillet) [inline packages/k-ramel/API.md]:
```suggestion
 - The second matcher is a function to ensure the request has not failed!
```

File: packages/k-ramel/API.md [hunk 6]
```diff
@@ -17,3 +17,83 @@ All options are optionals to keep the default usage as simple as possible.
 | hideRedux | `boolean` | `true` | actions and selectors from [`k-redux-factory`](https://github.com/alakarteio/k-redux-factory) are used without specifying `dispatch` or `getState`. |
 | enhancer | redux enhancer | `undefined` | usual `compose` and `applyMiddlare` you already use with Redux can be injected here (like router, redux-saga, etc). <br />`compose` and `applyMiddleware` from Redux are exposed by the lib. To use them:<br /> ```import { compose, applyMiddleware } from 'k-ramel'```. |
 | init | `object` | `undefined` | the default value of your store. |
+
+### types
+```js
+import { types } from 'k-ramel'
+```
+
+You can use when of our types as reducers (+ selectors and reactions).
```
Comment (delphinemillet) [inline packages/k-ramel/API.md]: I don't understand

Comment (fabienjuif) [reply packages/k-ramel/API.md]: one of

Comment (fabienjuif) [inline packages/k-ramel/API.md]:
```suggestion
You can use one of our types as reducers (+ selectors and reactions).
```

Comment (delphinemillet) [reply packages/k-ramel/API.md]: that's what I was thinking about ^^

**Comment by fabienjuif** [comment]
Thank you @delphinemillet

**Pull request merged**